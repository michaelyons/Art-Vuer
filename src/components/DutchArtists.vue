<template>
  <div class="hello">
    <h1>{{msg}}</h1>
    <article v-for="dutch in dutchArtists" :key="dutch.id">
      <h2>{{dutch.name}}</h2>
      <p>{{dutch.lifetime}}</p>
      <p>{{dutch.artwork}}</p>
      <button>View Artwork</button>
    </article>
  </div>
</template>

<script>
import GetDutchArtistsApi from '@/services/api/DutchArtists';
export default {
  name: 'DutchArtists',

  data() {
    return {
      dutchArtists: []
    };
  },
  created() {
    GetDutchArtistsApi.getDutchArtists()
      .then(dutchArtists => {
        this.dutchArtists = dutchArtists;
      })
      .catch(error => console.log(error));
  },
  mounted() {
    GetDutchArtistsApi.getDutchArtists()
      .then(dutchArtists => {
        const dutch = dutchArtists.map(artist => {
          return {
            name: artist.displayname,
            lifetime: artist.displaydate,
            artwork: artist.objectcount,
            website: artist.url,
            id: artist.id
          };
        });
        this.dutchArtists = dutch;
      })
      .catch(error => console.log(error));
  },
  props: {
    msg: String
  }
};
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
