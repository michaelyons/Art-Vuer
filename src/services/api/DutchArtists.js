import axios from "axios";

export default {
  getDutchArtists() {
    const key = process.env.VUE_APP_KEY;
    return axios
      .get(`/person?q=culture:Dutch&size=100&apikey=${key}`)
      .then(response => {
        return response.data.records;
      });
  }
};
