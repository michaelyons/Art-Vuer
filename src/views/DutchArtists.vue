<template>
  <div class="artists">
    <DutchArtists msg="Dutch Artists"/>
  </div>
</template>

<script>
import DutchArtists from '@/components/DutchArtists.vue';
export default {
  name: 'dutch-artists',
  components: {
    DutchArtists
  }
};
</script>
