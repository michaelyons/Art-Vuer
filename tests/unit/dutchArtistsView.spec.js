import { shallowMount } from "@vue/test-utils";
import DutchArtists from "@/views/DutchArtists.vue";

describe("DutchArtists.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(DutchArtists);
    expect(wrapper).toMatchSnapshot();
  });
});
