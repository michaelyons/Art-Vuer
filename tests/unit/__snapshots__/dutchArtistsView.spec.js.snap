// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DutchArtists.vue matches snapshot 1`] = `
<div class="artists">
  <dutchartists-stub msg="Dutch Artists"></dutchartists-stub>
</div>
`;
